core/libp2p/multi/multistream.hpp

```diff
@@ -6,14 +6,69 @@
 #ifndef KAGOME_MULTISTREAM_HPP
 #define KAGOME_MULTISTREAM_HPP
 
+#include <gsl/span>
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
 namespace libp2p::multi {
   /**
    * Format of stream identifier used in Libp2p
+   * @see https://github.com/multiformats/multistream
    */
   class Multistream {
+   public:
+    /**
+     * A protocol used in the stream is represented as a UNIX URI instead of just its name,
+     * since it's much more descriptive
+     */
+    using Path = std::string;
 
-  };
+    /**
+     * Creates an empty multistream
+     */
+    Multistream() noexcept = default;
```

Comment:
Are you sure we need this ctor? I don't see any applications of an empty `Multistream`

Reply:
As I remember, CI didn't want to build the code without this constructor because of some boost::variant magic.